evaluation_script/main.py

```diff
@@ -186,28 +196,36 @@ def evaluate(test_annotation_file, user_submission_file, phase_codename, **kwarg
     recall = 0
     fpr = 0
     f1 = 0
+    qa_precision = 0
+    qa_recall = 0
+    qa_fpr = 0
+    qa_f1 = 0
+
     if phase_codename == "unsupervised" or phase_codename == "supervised" or phase_codename == "final":
         print("evaluating for " +phase_codename+ " phase")
         extension = get_file_extension(user_submission_file)
         if extension == "tsv" or extension == "gz": 
             (tp, tn, fp, fn) = calculate_base_metrics(user_submission_file, truth)
-            precision = calculate_precision(tp, fp)
-            recall = calculate_recall(tp, fn)
-            fpr = calculate_recall(fp, tn)
-            f1 = calculate_f1(precision, recall)
-        else:
-            precision = 0
-            recall = 0
-            fpr = 0
-            f1 = 0
+            if not are_all_base_metrics_zero(tp, tn, fp, fn):
+                precision = calculate_precision(tp, fp)
+                recall = calculate_recall(tp, fn)
+                fpr = calculate_recall(fp, tn)
+                f1 = calculate_f1(tp, fp, fn)
+                (qa_precision, qa_recall, qa_fpr, qa_f1) = calculate_query_level_metrics()
+ 
         output["result"] = [
             {
                 "data": {
-                    "precision": precision,
-                    "recall": recall,
-                    "f1": f1,
-                    "tpr": recall,
-                    "fpr": fpr,
+                    "global_precision": precision,
+                    "global_recall": recall,
+                    "global_f1": f1,
+                    "global_tpr": recall,
+                    "global_fpr": fpr,
+                    "averaged_precision": qa_precision,
```

Comment:
Use "average" rather than "averaged". It's the more common term.

Reply:
will do

Reply:
Fixed in revision